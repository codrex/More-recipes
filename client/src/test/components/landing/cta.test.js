/* eslint-disable react/jsx-filename-extension */
import React from 'react';
import toJson from 'enzyme-to-json';
import CallToAction from '../../../components/pages/Landing/CallToAction';

const props = {
  push: jest.fn(),
  signin: jest.fn(),
  signup: jest.fn(),
};

describe('Cta form component ', () => {
  let component;
  beforeEach(() => {
    component = mount(<CallToAction {...props} />);
  });

  test('expect to match snapshot when signin button is clicked', () => {
    const tree = toJson(component);
    component.find('Button#login').simulate('click');
    expect(tree).toMatchSnapshot();
    expect(props.signin).toBeCalled();
    expect(tree).toBeInstanceOf(Object);
  });

  test('expect to match snapshot when signup button is clicked', () => {
    const tree = toJson(component);
    component.find('Button#signup').simulate('click');
    expect(tree).toMatchSnapshot();
    expect(props.signup).toBeCalled();
    expect(tree).toBeInstanceOf(Object);
  });
});
