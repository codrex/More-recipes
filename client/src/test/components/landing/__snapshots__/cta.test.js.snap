// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cta form component  expect to match snapshot when signin button is clicked 1`] = `
<CallToAction
  push={[Function]}
  signin={[Function]}
  signup={[Function]}
>
  <div
    className="cta "
  >
    <h1
      className="text-uppercase display-4 col-xs-11 col-sm-10 col-md-8 col-lg-9"
    >
      ​share​ and view ​awesome​ ​and​ ​exciting​ ​​recipes
    </h1>
    <div
      className="cta-btn-wrapper"
    >
      <Button
        Tag="button"
        className="btn-primary btn-lg login"
        dataTarget=""
        dataToggle=""
        handleClick={[Function]}
        href=""
        id="login"
        text="Login"
        type="button"
      >
        <button
          className="btn btn-primary btn-lg login"
          data-dismiss="modal"
          data-target=""
          data-toggle=""
          href=""
          id="login"
          onClick={[Function]}
          type="button"
        >
          Login
        </button>
      </Button>
      <Button
        Tag="button"
        className=" btn-secondary btn-lg reg text-white"
        dataTarget=""
        dataToggle=""
        handleClick={[Function]}
        href=""
        id="signup"
        text="Create an account"
        type="button"
      >
        <button
          className="btn  btn-secondary btn-lg reg text-white"
          data-dismiss="modal"
          data-target=""
          data-toggle=""
          href=""
          id="signup"
          onClick={[Function]}
          type="button"
        >
          Create an account
        </button>
      </Button>
    </div>
  </div>
</CallToAction>
`;

exports[`Cta form component  expect to match snapshot when signup button is clicked 1`] = `
<CallToAction
  push={[Function]}
  signin={[Function]}
  signup={[Function]}
>
  <div
    className="cta "
  >
    <h1
      className="text-uppercase display-4 col-xs-11 col-sm-10 col-md-8 col-lg-9"
    >
      ​share​ and view ​awesome​ ​and​ ​exciting​ ​​recipes
    </h1>
    <div
      className="cta-btn-wrapper"
    >
      <Button
        Tag="button"
        className="btn-primary btn-lg login"
        dataTarget=""
        dataToggle=""
        handleClick={[Function]}
        href=""
        id="login"
        text="Login"
        type="button"
      >
        <button
          className="btn btn-primary btn-lg login"
          data-dismiss="modal"
          data-target=""
          data-toggle=""
          href=""
          id="login"
          onClick={[Function]}
          type="button"
        >
          Login
        </button>
      </Button>
      <Button
        Tag="button"
        className=" btn-secondary btn-lg reg text-white"
        dataTarget=""
        dataToggle=""
        handleClick={[Function]}
        href=""
        id="signup"
        text="Create an account"
        type="button"
      >
        <button
          className="btn  btn-secondary btn-lg reg text-white"
          data-dismiss="modal"
          data-target=""
          data-toggle=""
          href=""
          id="signup"
          onClick={[Function]}
          type="button"
        >
          Create an account
        </button>
      </Button>
    </div>
  </div>
</CallToAction>
`;
